sbc_compassion/models/contracts.py

```diff
@@ -104,6 +97,32 @@ def _compute_write_for_birthday_alert(self):
                 or contract.last_letter > 90 - days_until_birthday
             )
 
+    ##########################################################################
+    #                              ORM METHODS                               #
+    ##########################################################################
+
+    @api.model
+    def create(self, vals):
+        sponsorship = super().create(vals)
+
+        # Set reading language by default for new sponsorships
+        correspondent_lang_id = False
+        for lang in sponsorship.correspondent_id.spoken_lang_ids:
+            if sponsorship.child_id.correspondence_language_id.id == lang.id:
+                self._update_reading_language(
+                    sponsorship.id,
+                    lang.id
+                )
+            if sponsorship.correspondent_id.lang == lang.lang_id.code:
+                correspondent_lang_id = lang.id
+        if not sponsorship.reading_language and correspondent_lang_id:
+            self._update_reading_language(
+                sponsorship.id,
+                correspondent_lang_id
+            )
+
+        return sponsorship
```

Comment:
You could do it before the actual create to avoid multiple sql transactions and simplify the code:

```suggestion
        if "child_id" in vals and "correspondent_id" in vals and "reading_language" not in vals:
            child = self.env["compassion.child"].browse(vals["child_id"])
            correspondent = self.env["res.partner"].browse(vals["correspondent_id"])
            correspondent_lang_id = False
            for lang in correspondent.spoken_lang_ids:
                if child.correspondence_language_id == lang:
                    vals["reading_language"] = lang.id
                    break
                if correspondent.lang == lang.lang_id.code:
                    correspondent_lang_id = lang.id
            else:
                vals["reading_language"] = correspondent_lang_id
        return super().create(vals)
```